layer/layer_store.go

```diff
@@ -106,11 +106,17 @@ func NewStoreFromGraphDriver(store MetadataStore, driver graphdriver.Driver) (St
 	return ls, nil
 }
 
-func (ls *layerStore) loadLayer(layer ChainID) (*roLayer, error) {
+func (ls *layerStore) loadLayer(layer ChainID) (l *roLayer, err error) {
 	cl, ok := ls.layerMap[layer]
 	if ok {
 		return cl, nil
 	}
+	defer func() {
+		// If failed to load the layer, remove the layer metadata.
+		if err != nil {
+			ls.store.Remove(layer)
+		}
+	}()
```

Comment:
`loadLayer` loads the layer metadata from disk, if failure means something wrong with the metadata. I think if we do not remove the data on disk , could cause the layer data in memory and disk do not keep the same. And also cause the  issue as collljt0725 mentioned.